server/model/monitor.js

```diff
@@ -23,6 +22,8 @@ const Gamedig = require("gamedig");
 const jsonata = require("jsonata");
 const jwt = require("jsonwebtoken");
 const Database = require("../database");
+const { CookieJar } = require("tough-cookie");
```

Comment:
I don't quite understand how adding these two dependencies solves the problem.
- Why is `http-cookie-agent` needed? 
  (what is the difference between `https.Agent` and `HttpsCookieAgent`)
- How/is the `CookieJar` shared?
  (Can other requests leak cookies via this mechanism?)

Reply:
`tough-cookie` should be just a pure standalone cookie api and storage. `http-cookie-agent` connects between `tough-cookie` and `axios`.

However, the problem is that a http agent and `axios` is an one-to-one relationship and it should be used by `CacheableDnsHttpAgent` already. 

And the fun fact is that I just realized that I maybe messed up something in #1853, `CacheableDnsHttpAgent` does not seem to be used in http/https monitor.

Reply:
A new cookie jar is created for every request. It should be per request meaning that no cookies are ever shared outside of a single request if my understanding of the `beat` function is correct. It was my understanding that the `CachableDnsHttpAgent` was only used for docker and steam types of heartbeats meaning there shouldn't be any conflict since the beat types are mutually exclusive (else if). Those types also don't need cookie storage so I thought it wasn't needed there. I do think though there should be one central place to create an http agent rather than many things each creating one slightly differently but that's more a matter of design. I tested the changes and they worked as expected. Should I implement it differently?

Reply:
I also must be missing something because I don't see any references for `CacheableDnsHttpAgent.registerGlobalAgent()` in the project in the `2.0.X` branch or the `master` branch.